src/amo/reducers/reviews.js

```diff
@@ -313,6 +316,21 @@ export default function reviewsReducer(
     }
     case SET_REVIEW: {
       const { payload } = action;
+      const review = createInternalReview(payload);
+
+      return {
+        ...state,
+        byId: storeReviewObjects({ state, reviews: [review] }),
+        byUserId: {
```

Comment:
It's a shame that this is a duplication of the `SET_INTERNAL_REVIEW` case. However, I don't see a clean way to DRY it up. People complain about how Redux paradigms can be hard to encapsulate in abstractions and this is one such example.

Unless you have an idea for how to abstract this cleanly, you'll need to add test coverage for the `SET_INTERNAL_REVIEW` case by copying [this test](https://github.com/mozilla/addons-frontend/blob/5cdc08bf0cccb083f487aed8de70e449b37cd0b6/tests/unit/amo/reducers/testReviews.js#L109). If I comment it out in your current patch, the tests pass.

Reply:
I could extract the common logic into a function at least.

I tried to use the same action, looking for `userId` in the `payload` to determine whether `createInternalReview()` was needed, but Flow complained.